examples/chip-tool/main.cpp

```diff
@@ -88,12 +89,57 @@ static void OnConnect(DeviceController::ChipDeviceController * controller, Trans
     }
 }
 
+static bool ContentMayBeADataModelMessage(System::PacketBuffer * buffer)
+{
+    // A data model message has a first byte whose value is always one of  0x00,
+    // 0x01, 0x02, 0x03.
+    return buffer->DataLength() > 0 && buffer->Start()[0] < 0x04;
+}
+
+// This function consumes (i.e. frees) the buffer.
+static void HandleDataModelMessage(System::PacketBuffer * buffer)
+{
+    EmberApsFrame frame;
+    bool ok = extractApsFrame(buffer->Start(), buffer->DataLength(), &frame);
```

Comment:
VerifyOrExit() would work here, too instead of an early return?

Reply:
This is "ExtractDataModelMessage()" or maybe "Parse"?

Reply:
That doesn't compile, because it involves a goto across the initialization of `messageLen`.  I could probably make it work with enough contortion, but I'm not sure it's worth it.